indexing.go

```diff
@@ -105,33 +105,81 @@ func BulkIndex(docs []string, options Options) error {
 				return err
 			}
 
-			// Find ID in the document.
-			id, ok := docmap[options.IDField]
-			if !ok {
-				return fmt.Errorf("document has no ID field (%s): %s", options.IDField, doc)
-			}
-
+			idstring := options.IDField                            //A delimiter separates string with all the fields to be used as ID
+			id:=strings.FieldsFunc(idstring, func (r rune) bool { return r == ',' || r == ' '})			
+		//	idtemp := strings.Split(idstring, ",")
+		//	id := strings.Split(idtemp, " ")
 			// ID can be any type at this point, try to find a string
 			// representation or bail out.
 			var idstr string
-			switch t := id.(type) {
-			case string:
-				idstr = t
-			case fmt.Stringer:
-				idstr = t.String()
-			case json.Number:
-				idstr = t.String()
-			default:
-				return fmt.Errorf("cannot convert %T id value to string: %v", id, id)
+			var currrentId string
+			for counter:= range id{
+				currrentId = id[counter]
+				tokstr := strings.Split(currrentId,".")
+				if len(tokstr)>1{
```

Comment:
`len(tokstr)>1` indicates that you did not run [gofmt](https://golang.org/cmd/gofmt/) (or [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), respectively) on your code. Please do that, even for a PR draft (written in Go). Most editors have support for applying `gofmt` or `goimports` automatically on save.